packages/curve-ui-kit/src/features/connect-wallet/lib/useWallet.ts

```diff
@@ -38,14 +38,7 @@ const useWallet = () => {
 
   // use the async functions so we can properly handle the promise failures. We could instead use query state in the future.
   const { connectAsync } = useConnect()
-  const { disconnectAsync } = useDisconnect()
-
-  const disconnect = useCallback(async () => {
-    // Call disconnectAsync twice to ensure the useAccount hooks properly reset
-    // Bro I don't even know why, but if you call it once the useAccount states won't update...
-    await disconnectAsync()
-    await disconnectAsync()
-  }, [disconnectAsync])
+  const { disconnect } = useDisconnect()
```

Comment:
We should be displaying connect/disconnect errors

Reply:
Connect errors are already shown, not sure where to place disconnect errors that isn't a snackbar / toast.

Reply:
Toast should be good enough